### PR TITLE
Getting rid of more docs warnings

### PR DESCRIPTION
**Pull request by jsignell**
- [x] Tests passed
- [x] Passes `black dask` / `flake8 dask`

This PR brings the warnings down, but I'm still working on trying to get rid of all of them.

### CODE REVIEW

**Comment by jsignell** [comment]
Actually @jrbourbeau are you good with merging this as is? I've realized I should chip away at big tasks like this and merge PRs as I go rather than letting them sit and grow stale.

File: dask/array/reductions.py [hunk 1]
```diff
@@ -451,74 +451,74 @@ def nansum(a, axis=None, dtype=None, keepdims=False, split_every=None, out=None)
     )
 
 
-with ignoring(AttributeError):
```
Comment (jrbourbeau) [inline dask/array/reductions.py:454]: Why remove this `ignoring(AttributeError)`? Is this a NumPy version thing we no longer need?

**Review by jrbourbeau — APPROVED** [review]
Could you clarify why we're removing a few `with ignoring(AttributeError):`? Otherwise this looks good to merge

File: dask/array/reductions.py (hunk 1 above)
Comment (jsignell) [reply dask/array/reductions.py:454]: Yeah that was my read of it. It was added 6 years ago and seems to work fine with our oldest supported version of numpy.

Comment (jrbourbeau) [reply dask/array/reductions.py:454]: Great, thanks for clarifying 👍

**Pull request merged**